demo-project/run.sh

```diff
@@ -4,7 +4,13 @@ cd .. && ./gradlew clean publishAllPublicationsToLocalRepository
 
 cd demo-project
 
-./gradlew -I ../reference/configure-develocity.gradle \
+mkdir -p build
+rm build/configure-develocity.gradle
+echo 'initscript { repositories { maven { url = "../build/local-repo" } } }' >> build/configure-develocity.gradle
+echo '' >> build/configure-develocity.gradle
+cat ../reference/configure-develocity.gradle >> build/configure-develocity.gradle
```

Comment:
This logic feels like it should be in a Gradle task rather than a bash script.
Then we don't need to call `run.sh` in the action, we can simply execute the same task.